### PR TITLE
Graphql api [PLOPS-253]

### PR DESCRIPTION
**Pull request by mikemonkers**

### CODE REVIEW

File: cloudflare-analytics.sh [hunk 1]
```diff
@@ -36,46 +36,61 @@ cloudflareapikey="${CLOUDFLARE_API_TOKEN:="dummy_api_key"}"
...
+CF_DEFAULT_SINCE_MIN="-10080"
+CF_DEFAULT_UNTIL_MIN="0"
+
 ######################################
 # Define this to be "echo" in ENV to switch off non-observer operations.
 : "${ECHO:=""}"
 
-function process_results {
+#####################################################
+# Processing Analytics API results.
```
Comment (dannystaple) [inline cloudflare-analytics.sh:55]: Is it work stating params in the block comment - there's 2 here - say what they mean perhaps?

File: cloudflare-analytics.sh [hunk 2]
```diff
@@ -115,114 +130,393 @@ bandwidth=$bandwidth"
...
+
+    ## Unique visits
+    cfUniquesAll=$(echo "$output" | jq --raw-output ".uniq.uniques")
+
+    ## Timestamp
+    date=$(echo "$output" | jq --raw-output ".dimensions.datetime")
+    cfTimeStamp=$(date -d "${date}" '+%s')
+
+# The alignment here is important (DO NOT INDENT)
+    series_name="cloudflare_analytics"
```
Comment (dannystaple) [inline cloudflare-analytics.sh:193]: If this stuff below is essentially the same as line 121 - why repeat it?

Comment (mikemonkers) [reply cloudflare-analytics.sh:193]: Well... whilst I see you point... this is in essence a 1 line string assignment. Creating a function to pass the params in just to assign a string seems excessive.

But also I imagine the 'api' variant will be removed when that api is deprecated (or sooner if we are happy with the new one).

File: cloudflare-analytics.sh (hunk 1 above)
Comment (mikemonkers) [reply cloudflare-analytics.sh:55]: Actually, the second param should. be removed as it isn't used. But you missed that right? ;-)

Comment (mikemonkers) [reply cloudflare-analytics.sh:55]: Which is now done.

Comment (dannystaple) [reply cloudflare-analytics.sh:55]: Think this should be something other than output here.

File: cloudflare-analytics.sh (hunk 2 above)
Comment (dannystaple) [reply cloudflare-analytics.sh:193]: Resolved

File: cloudflare-analytics.sh [hunk 3]
```diff
@@ -36,46 +36,58 @@ cloudflareapikey="${CLOUDFLARE_API_TOKEN:="dummy_api_key"}"
...
+      process_request_api "$values"
     done
 }
 
-function process_result {
+function process_request_api {
     local output="$1"
-    local index=${2:0}
-    local data_tmp="${cloudflarezone}_data.$$.tmp"
+    local data_tmp="api_$TMP_FILE_POSTFIX"
```
Comment (dannystaple) [inline cloudflare-analytics.sh]: Should this be using mktemp? Should end up in the right place.

File: cloudflare-analytics.sh [hunk 4]
```diff
@@ -36,46 +36,58 @@ cloudflareapikey="${CLOUDFLARE_API_TOKEN:="dummy_api_key"}"
 cloudflarezoneid="${CLOUDFLARE_ZONE_ID:="dummy_zone_id"}"
 cloudflarezone="${CLOUDFLARE_ZONE_NAME:="dummy_zone_name"}"
 
+TMP_FILE_LASTRUN_TIME="${cloudflarezone}_lastrun.tmp"
```
Comment (dannystaple) [inline cloudflare-analytics.sh]: This needs to be in the tmp filesystem.

File: cloudflare-analytics.sh [hunk 5]
```diff
@@ -115,114 +127,391 @@ bandwidth=$bandwidth"
...
+            }
+          }
+        }
+      }\",
+      \"variables\": $gql_variables
+    }"
+    make_graphql_post "$payload"
+}
+
+#function fetch_lb_data_graphql {
```
Comment (dannystaple) [inline cloudflare-analytics.sh:439]: dead code? or todo?

Comment (mikemonkers) [reply cloudflare-analytics.sh:439]: Potential ToDo. I shall comment accordingly.

File: cloudflare-analytics.sh (hunk 3 above)
Comment (mikemonkers) [reply cloudflare-analytics.sh]: Sure.... done

File: cloudflare-analytics.sh (hunk 4 above)
Comment (mikemonkers) [reply cloudflare-analytics.sh]: Done

File: cloudflare-analytics.sh (hunk 5 above)
Comment (mikemonkers) [reply cloudflare-analytics.sh:439]: Done

File: cloudflare-analytics.sh (hunk 1 above)
Comment (dannystaple) [reply cloudflare-analytics.sh:55]: We've discussed this - going to set it to resolved for now.

**Review by dannystaple — APPROVED** [review]

**Review by dannystaple — APPROVED** [review]
Still looks good to me

**Review by chrisedebo — APPROVED** [review]
Looks good to me 👍

**Pull request merged**